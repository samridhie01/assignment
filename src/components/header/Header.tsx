import React from "react";
import AppBar from "@material-ui/core/AppBar";
import {createStyles, IconButton, makeStyles, Theme, Toolbar} from "@material-ui/core";
import logo from "../../assets/planet2.svg";
import Typography from "@material-ui/core/Typography";
import {Link} from "react-router-dom";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        menuButton: {
            marginRight: theme.spacing(2),
        },
        title: {
            flexGrow: 1,
        },
        logo: {
            width: 50,
            height: 50
        },
    }),
);

interface HeaderProps {
    path: string,
    linkName: string
}

const Header: React.FC<HeaderProps> = ({
     path,
     linkName
     }) => {
    const classes = useStyles();
    return (
        <AppBar color="inherit" position="static" data-testid={"header-section"}>
            <Toolbar>
                <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                    <img src={logo} style={{height: 100, width: 100}} alt="website logo"/>
                </IconButton>
                <Typography variant="h6" className={classes.title}>
                    Planets
                </Typography>
                <Link to={{pathname: `${path}`}} color="inherit">{linkName}</Link>
            </Toolbar>
        </AppBar>
    )


}

export default Header;
