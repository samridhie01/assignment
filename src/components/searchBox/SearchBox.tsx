import React, { useState } from "react";
import {
  Button,
  Box,
  TextField,
  Theme,
  makeStyles,
  createStyles
} from "@material-ui/core";
import {getSearchKey} from "../utils/utils";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    alignCenter:{
      alignSelf: "center",
      width: "50%"
    },
    customStyle:{
     width: 350,
      padding: "20px"
    }
  })
);

interface Search {
  handleSearch: () => void;
}


const SearchBox: React.FC<Search> = ({ handleSearch }) => {
  const [searchItem, setSearchItem] = useState<string>("");
  const handleSearchChange = (event: any) => {
    setSearchItem(event.target.value);
    localStorage.setItem("searchKey", event.target.value);
  };

  const handleSearchButtonClick =() =>{
    handleSearch();
  }

  const classes = useStyles();

  return (
    <>
      <div >
      <Box m={4} border={1} width={400} height={150} display="flex" flexDirection={"column"} data-testid={"search-section"}>
        <TextField
          variant="outlined"
          onChange={handleSearchChange}
          placeholder="Enter climate to filter planets"
          className={classes.customStyle}
          fullWidth={false}
          inputProps={{ "data-testid": "climate-filter" }}
          value={getSearchKey() || ''}
        />
        <Button
          variant="contained"
          onClick={handleSearchButtonClick}
          fullWidth={false}
          data-testid={'searchBtn'}
          className={classes.alignCenter}
        >
          Search
        </Button>
        </Box>
      </div>
    </>
  );
};

export default SearchBox;
