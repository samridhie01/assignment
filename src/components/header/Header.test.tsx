import { render,  } from '@testing-library/react';
import React from "react";
import Header from "./Header";
import SearchBox from "../searchBox/SearchBox";
import {BrowserRouter} from "react-router-dom";


describe("Header", ()=>{
    it('should render snapshot', () => {

        const { asFragment } = render(
            <BrowserRouter>
            <Header path="/favs" linkName={"Favorites"}/>
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

