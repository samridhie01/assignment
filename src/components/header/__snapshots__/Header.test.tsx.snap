// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorInherit MuiPaper-elevation4"
    data-testid="header-section"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="website logo"
            src="planet2.svg"
            style="height: 100px; width: 100px;"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
      >
        Planets
      </h6>
      <a
        color="inherit"
        href="/favs"
      >
        Favorites
      </a>
    </div>
  </header>
</DocumentFragment>
`;
